client/FileUpload.js

```diff
@@ -54,6 +54,28 @@ class FileUpload extends Component {
             }
             this.setState({fileDoc: fileDoc});
         };
+
+        this.handleLinkBLur = event => {
+            if (this.props.fileType === 'Video') {
+                let fileDoc = this.state.fileDoc;
+                let url = event.target.value;
+                if (!url.includes('youtu.be/')) { // Not a sharable link
+                    if (url.includes('youtube.com/watch?v='))  {
+                        // ...but we may be able to fix it
+                        let videoId;
+                        if (url.includes('&')) { // link includes extra params - remove them
```

Comment:
this doesn't account for if `v=` is not first param

Reply:
You're right, I completely missed that. Fixed

Reply:
You can use regex to get video id, so you don't have to worry about covering all the edge cases. There can be any number of get params in video (maybe user has browser ext, etc)
```
let regex = /\?.*v=([^&]*)(\&|$)/gi;
let match = regex.exec(url);
let id=match[0];
```

Reply:
Had to fix a bug in the regex, but was able to swap my previous code for it. Thanks!